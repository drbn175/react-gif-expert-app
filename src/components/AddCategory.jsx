import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory= ({onNewCategory}) =>{

    const [inputValue, setInputValue] = useState();

    const onInputChange = ({target}) =>{
        setInputValue(target.value);
    }

    const onSubmit = (event) =>{
        event.preventDefault();
        if(inputValue.trim().length <= 2 ) return;
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return(
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Search gif"
                value = { inputValue }
                onChange={onInputChange}
            ></input>
        </form>
    )
}

AddCategory.propTypes={
    onNewCategory: PropTypes.func.isRequired
}