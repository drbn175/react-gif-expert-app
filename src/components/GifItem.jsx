export const GifItem = ({title, url, id}) => {
    return (
        <div className="card" key={id}>
            <img src={url} alt={{title}}></img>
            <p>{title}</p>
        </div>
        )
}