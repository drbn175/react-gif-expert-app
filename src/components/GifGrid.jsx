import { useFetchGifs } from "../hooks/UseFetchGifs";
import { GifItem } from "./GifItem";


export const GifGrid = ({ category}) =>{

    const {images, isLoading} = useFetchGifs(category);

    return (
        <>
        
        <div key={category}>
            <h3>{ category }</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }
            <div className="card-grid">
            {
                images.map((img) => (
                    <GifItem key={img.id} 
                    {...img} />
                ))
            }
            </div>
           
        </div>
        </>
    )
}