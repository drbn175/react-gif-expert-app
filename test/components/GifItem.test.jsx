import { GifItem } from "../../src/components/GifItem";
import { render, screen } from "@testing-library/react";

describe('Test GifItem', () =>{

    const title = 'Test';
    const url = 'https://www.google.com';
    test('should match snapshot',()=>{
        const {container} = render( <GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });


    test('should show img with url', ()=>{
        render( <GifItem title={title} url={url} />);
        const {src, alt} = screen.getByRole('img');
        expect( src).toBe(src);
        expect( alt).toBe(alt);
    });

    test('should show title ', ()=>{
        render( <GifItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();
    })
});