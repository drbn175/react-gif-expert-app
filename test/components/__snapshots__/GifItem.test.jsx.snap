// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="[object Object]"
      src="https://www.google.com"
    />
    <p>
      Test
    </p>
  </div>
</div>
`;
